maestro-orchestra-models/src/main/java/maestro/orchestra/Commands.kt

```diff
@@ -426,3 +426,37 @@ data class SetLocationCommand(
         return this
     }
 }
+
+data class RepeatCommand(
+    val times: String? = null,
+    val condition: Condition? = null,
+    val commands: List<MaestroCommand>,
+) : CompositeCommand {
+
+    override fun subCommands(): List<MaestroCommand> {
+        return commands
+    }
+
+    override fun description(): String {
+        val timesInt = times?.toIntOrNull() ?: 1
+
+        return when {
+            condition != null && timesInt > 1 -> {
+                "Repeat while ${condition.description()} (up to $timesInt times)"
+            }
+            condition != null -> {
+                "Repeat while ${condition.description()}"
```

Comment:
Curious, do we have complementary tests for repeat under a certain `condition`?

Reply:
We decided to remove `while` condition for now (removed [here](https://github.com/mobile-dev-inc/maestro/pull/255/commits/3b8ae1ad670eb471deabc38107560d1f630a01f6)) until we are absolutely sure that it is needed

Reply:
Got it, thank you.